app/scripts/controllers/app-metadata.test.ts

```diff
@@ -0,0 +1,205 @@
+import assert from 'assert';
+import { AppMetadataController } from './app-metadata';
+
+const MOCK_VERSION = {
+  version: '',
+};
+
+const EXPECTED_DEFAULT_STATE = {
+  currentAppVersion: '',
+  previousAppVersion: '',
+  previousMigrationVersion: 0,
+  currentMigrationVersion: 0,
+};
+
+jest.mock(
+  '../platforms/extension',
+  () =>
+    class MockExtensionPlatform {
+      getVersion(): string {
+        return MOCK_VERSION.version;
+      }
+    },
+);
+
+describe('NetworkController', () => {
+  describe('constructor', () => {
+    it('accepts initial state and does not modify it if currentMigrationVersion and platform.getVersion() match respective values in state', async () => {
+      const initState = {
+        currentAppVersion: '1',
+        previousAppVersion: '1',
+        previousMigrationVersion: 1,
+        currentMigrationVersion: 1,
+      };
+      MOCK_VERSION.version = initState.currentAppVersion;
+      const appMetadataController = new AppMetadataController({
+        state: initState,
+        currentMigrationVersion: 1,
+      });
+      assert.deepStrictEqual(appMetadataController.store.getState(), initState);
+    });
+
+    it('sets default state and does not modify it if currentMigrationVersion and platform.getVersion() match respective default values', async () => {
+      MOCK_VERSION.version = '';
+      const appMetadataController = new AppMetadataController({
+        state: {},
+        currentMigrationVersion: 0,
+      });
+      assert.deepStrictEqual(
+        appMetadataController.store.getState(),
+        EXPECTED_DEFAULT_STATE,
+      );
+    });
+
+    it('updates the currentAppVersion state property via a call to maybeUpdateAppVersion if platform.getVersion() does not match the default value', async () => {
```

Comment:
Better to not mention the implementation in the test description. The implementation isn't relevant here, and it adds further maintenance burden if it is mentioned.
```suggestion
    it('updates the currentAppVersion state property if platform.getVersion() does not match the default value', async () => {
```

Reply:
Same for the following tests as well

Reply:
good suggestion, thanks

addressed in 1b88d8c7ed